src/search/landmarks/landmark_status_manager.h

```diff
@@ -9,31 +9,45 @@ namespace landmarks {
 class LandmarkGraph;
 class LandmarkNode;
 
-enum landmark_status {lm_reached = 0, lm_not_reached = 1, lm_needed_again = 2};
+enum LandmarkStatus {PAST = 0, FUTURE = 1, PAST_AND_FUTURE = 2};
 
 class LandmarkStatusManager {
     LandmarkGraph &lm_graph;
+    const bool progress_goals;
+    const bool progress_greedy_necessary_orderings;
+    const bool progress_reasonable_orderings;
 
-    PerStateBitset reached_lms;
-    std::vector<landmark_status> lm_status;
+    PerStateBitset past_landmarks;
+    PerStateBitset future_landmarks;
+    std::vector<LandmarkStatus> lm_status;
 
-    bool landmark_is_leaf(const LandmarkNode &node, const BitsetView &reached) const;
-    bool landmark_needed_again(int id, const State &state);
-
-    void set_reached_landmarks_for_initial_state(
+    void progress_initial_state(
         const State &initial_state, utils::LogProxy &log);
+    void progress_basic(
+        const BitsetView &parent_past, const BitsetView &parent_fut,
+        const State &parent_ancestor_state, BitsetView &past, BitsetView &fut,
+        const State &ancestor_state);
+    void progress_goal(int id, const State &ancestor_state, BitsetView &fut);
+    void progress_greedy_necessary_ordering(int id, const State &ancestor_state,
+                                            const BitsetView &past, BitsetView &fut);
+    void progress_reasonable_ordering(int id, const BitsetView &past, BitsetView &fut);
 public:
-    explicit LandmarkStatusManager(LandmarkGraph &graph);
+    LandmarkStatusManager(
+        LandmarkGraph &graph,
+        bool progress_goals,
+        bool progress_greedy_necessary_orderings,
+        bool progress_reasonable_orderings);
 
-    BitsetView get_reached_landmarks(const State &state);
+    BitsetView get_past_landmarks(const State &state);
+    BitsetView get_future_landmarks(const State &state);
 
     void update_lm_status(const State &ancestor_state);
 
     void process_initial_state(
         const State &initial_state, utils::LogProxy &log);
-    bool process_state_transition(
```

Comment:
Looking at the implementation, the return value seems to be `false` if the transition induces a self-loop and `true` otherwise. I'm not sure where this information could come in handy, but given that (a) it was never used and (b) it is nowhere documented, I think it's fine to drop the return value entirely.